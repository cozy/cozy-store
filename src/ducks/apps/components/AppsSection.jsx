import React from 'react'
import { translate } from 'cozy-ui/react/I18n'
import withBreakpoints from 'cozy-ui/react/helpers/withBreakpoints'
import { getTranslatedManifestProperty } from 'lib/helpers'

import SmallAppItem from 'ducks/components/SmallAppItem'

const _renderAppComponent = (app, t, onAppClick, isMobile) => {
  return (
    <SmallAppItem
      app={app}
      namePrefix={getTranslatedManifestProperty(app, 'name_prefix', t)}
      name={getTranslatedManifestProperty(app, 'name', t)}
      onClick={() => onAppClick(app.slug)}
      key={app.slug}
      isMobile={isMobile}
    />
  )
}

export const AppsSection = ({
  t,
  appsList,
  subtitle,
  onAppClick,
  breakpoints = {}
}) => {
  const { isMobile } = breakpoints
  return (
    <div className="sto-sections-apps">
      {subtitle}
      {appsList && !!appsList.length && (
        <div className="sto-sections-list">
          {appsList.map(app =>
            _renderAppComponent(app, t, onAppClick, isMobile)
          )}
        </div>
      )}
    </div>
  )
}

export default translate()(withBreakpoints()(AppsSection))
