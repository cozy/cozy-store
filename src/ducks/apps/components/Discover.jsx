import { useAlternativeStore } from 'ducks/AlternativeStore/useAlternativeStore'
import ApplicationRouting from 'ducks/apps/components/ApplicationRouting'
import Sections from 'ducks/apps/components/QuerystringSections'
import AppsLoading from 'ducks/components/AppsLoading'
import { useNavigateNoUpdates, withRouterUtils } from 'lib/RouterUtils'
import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { useMatch, useSearchParams } from 'react-router-dom'

import { BarCenter } from 'cozy-bar'
import { Content } from 'cozy-ui/transpiled/react/Layout'
import Typography from 'cozy-ui/transpiled/react/Typography'
import withBreakpoints from 'cozy-ui/transpiled/react/helpers/withBreakpoints'
import { translate } from 'cozy-ui/transpiled/react/providers/I18n'

export class Discover extends Component {
  constructor(props) {
    super(props)
    this.onAppClick = this.onAppClick.bind(this)
  }

  onAppClick(appSlug) {
    const { navigate, searchParams } = this.props
    const search = searchParams.size > 0 ? `?${searchParams.toString()}` : ''
    navigate(`/discover/${appSlug}${search}`)
  }

  render() {
    const {
      t,
      apps,
      isFetching,
      isAppFetching,
      fetchError,
      isUninstalling,
      actionError,
      breakpoints = {},
      isExact,
      intentData,
      onTerminate
    } = this.props

    const { isMobile } = breakpoints

    return (
      <Content>
        {isExact && isFetching && <AppsLoading />}
        <div>
          {isMobile && !intentData && (
            <BarCenter>
              <Typography component="h2" variant="h4">
                {t('discover.title')}
              </Typography>
            </BarCenter>
          )}
          <div>
            {!isFetching && (
              <Sections
                apps={apps}
                error={fetchError}
                onAppClick={this.onAppClick}
                intentData={intentData}
                parent="discover"
              />
            )}
          </div>
        </div>

        <ApplicationRouting
          apps={apps}
          isFetching={isFetching}
          isAppFetching={isAppFetching}
          isUninstalling={isUninstalling}
          actionError={actionError}
          intentData={intentData}
          onTerminate={onTerminate}
          parent="discover"
        />
      </Content>
    )
  }
}

const DiscoverWrapper = props => {
  const isExact = useMatch('discover')
  const navigate = useNavigateNoUpdates()
  const [searchParams] = useSearchParams()
  const { alternativeApps } = useAlternativeStore()

  return (
    <Discover
      {...props}
      apps={[...alternativeApps, ...props.apps]}
      isExact={isExact}
      navigate={navigate}
      searchParams={searchParams}
    />
  )
}

DiscoverWrapper.propTypes = {
  apps: PropTypes.array,
  isFetching: PropTypes.bool.isRequired,
  isAppFetching: PropTypes.bool.isRequired,
  isUninstalling: PropTypes.bool.isRequired,
  fetchError: PropTypes.object,
  actionError: PropTypes.object,
  intentData: PropTypes.shape({
    appData: PropTypes.object,
    data: PropTypes.object
  }),
  onTerminate: PropTypes.func,
  /* With HOC */
  breakpoints: PropTypes.object.isRequired,
  t: PropTypes.func.isRequired
}

export default translate()(withBreakpoints()(withRouterUtils(DiscoverWrapper)))
