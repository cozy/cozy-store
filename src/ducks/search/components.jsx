import StoreAppItem from 'ducks/apps/components/StoreAppItem'
import { dumpMatches } from 'ducks/search/utils'
import sortBy from 'lodash/sortBy'
import PropTypes from 'prop-types'
import React, { useCallback, useMemo } from 'react'

import flag from 'cozy-flags'
import { useI18n } from 'cozy-ui/transpiled/react/I18n'
import Icon from 'cozy-ui/transpiled/react/Icon'
import Magnifier from 'cozy-ui/transpiled/react/Icons/Magnifier'
import Input from 'cozy-ui/transpiled/react/Input'
import InputGroup from 'cozy-ui/transpiled/react/InputGroup'
import Label from 'cozy-ui/transpiled/react/Label'
import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'

export const SearchField = ({ onChange, value }) => {
  const { t } = useI18n()
  const { isMobile } = useBreakpoints()
  const handleChange = useCallback(
    ev => {
      onChange(ev.target.value)
    },
    [onChange]
  )
  return (
    <div className="u-flex u-flex-items-center u-flex-grow-1">
      {!isMobile ? (
        <Label className="u-di u-mr-half" htmlFor="discover-search">
          {t('discover-search-field.label')}
        </Label>
      ) : null}
      <InputGroup
        prepend={
          isMobile ? (
            <Icon icon={Magnifier} className="u-pl-1 u-coolGrey" />
          ) : null
        }
      >
        <Input
          id="discover-search"
          placeholder={t('discover-search-field.placeholder')}
          onChange={handleChange}
          type="text"
          value={value}
        />
      </InputGroup>
    </div>
  )
}

export const SearchResults = ({ searchResults, onAppClick, disableClick }) => {
  const sortedSortResults = useMemo(() => {
    return sortBy(searchResults, result => result.score)
  }, [searchResults])
  return (
    <div className="u-mv-1 u-flex u-flex-wrap">
      {sortedSortResults.map(result => {
        const app = result.item
        const isDisableClick = disableClick?.(app)
        return flag('store.show-search-score') ? (
          <div>
            <StoreAppItem
              onClick={() => !isDisableClick && onAppClick(app.slug)}
              key={app.slug}
              app={app}
            />
            <small
              onClick={() => dumpMatches(result)}
              title={JSON.stringify(result.matches, null, 2)}
            >
              {result.score.toFixed(2)}
            </small>
          </div>
        ) : (
          <StoreAppItem
            onClick={() => !isDisableClick && onAppClick(app.slug)}
            key={app.slug}
            app={app}
          />
        )
      })}
    </div>
  )
}

SearchResults.propTypes = {
  onAppClick: PropTypes.func.isRequired,
  disableClick: PropTypes.func,
  searchResults: PropTypes.array
}
