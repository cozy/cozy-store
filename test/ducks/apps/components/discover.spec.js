'use strict'

/* eslint-env jest */
/* global cozy */

import React from 'react'
import { shallow } from 'enzyme'

import { tMock } from 'jestLib/I18n'
import { Discover } from 'ducks/apps/components/Discover'

import mockApps from 'ducks/apps/_mockApps'

const { BarCenter } = cozy.bar

const mockError = new Error('This is a test error')

const mockRegistyApps = mockApps
  .filter(app => app.isInRegistry)
  .filter(
    app => Array.isArray(app.versions.stable) && !!app.versions.stable.length
  )

const getMockProps = (
  apps = mockRegistyApps,
  isFetching = false,
  fetchError = null
) => ({
  fetchApps: jest.fn(),
  apps,
  isFetching,
  fetchError,
  actionError: null,
  navigate: jest.fn()
})

describe('Discover component', () => {
  it('should be rendered correctly with apps', () => {
    const mockProps = getMockProps()
    const component = shallow(
      <Discover t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should be rendered correctly with apps and URL search params', () => {
    const mockProps = getMockProps(
      mockRegistyApps,
      false,
      null,
      { isExact: true },
      { search: '?category=cozy' }
    )
    const component = shallow(
      <Discover t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should render correctly a spinner if apps is fetching', () => {
    const mockProps = getMockProps([], true, null)
    const component = shallow(
      <Discover t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should display error from props correctly', () => {
    const mockProps = getMockProps([], false, mockError)
    const component = shallow(
      <Discover t={tMock} {...mockProps} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should define the correct onAppClick function to pass to sections', () => {
    const mockProps = getMockProps()
    const component = shallow(<Discover t={tMock} {...mockProps} />)
    const instance = component.instance()
    instance.onAppClick(mockRegistyApps[0].slug)
    expect(mockProps.navigate).toHaveBeenCalledTimes(1)
    expect(mockProps.navigate).toHaveBeenCalledWith(`/discover/collect`)
  })

  it('should define the correct onAppClick function to pass to sections with redirectAfterInstall search params, and so replace __APPSLUG__', () => {
    const mockProps = getMockProps()
    const component = shallow(
      <Discover
        t={tMock}
        {...mockProps}
        searchParams={{
          get: () =>
            'http://mespapiers.mycozy.cloud/#/paper/files/energy_invoice/harvest/'
        }}
      />
    )
    const instance = component.instance()
    instance.onAppClick(mockRegistyApps[0].slug)
    expect(mockProps.navigate).toHaveBeenCalledTimes(1)
    expect(mockProps.navigate).toHaveBeenCalledWith(
      `/discover/collect?redirectAfterInstall=http%3A%2F%2Fmespapiers.mycozy.cloud%2F%23%2Fpaper%2Ffiles%2Fenergy_invoice%2Fharvest%2F%3FconnectorSlug%3Dcollect`
    )
  })

  it('should encode uri when using redirectAfterInstall search params', () => {
    const mockProps = getMockProps()
    const component = shallow(
      <Discover
        t={tMock}
        {...mockProps}
        searchParams={{
          get: () => 'http://mydomain/#/paper?param1=val1&param2=val2'
        }}
      />
    )
    const instance = component.instance()
    instance.onAppClick(mockRegistyApps[0].slug)
    expect(mockProps.navigate).toHaveBeenCalledTimes(1)
    expect(mockProps.navigate).toHaveBeenCalledWith(
      `/discover/collect?redirectAfterInstall=http%3A%2F%2Fmydomain%2F%23%2Fpaper%3Fparam1%3Dval1%26param2%3Dval2%3FconnectorSlug%3Dcollect`
    )
  })

  it('should use BarCenter from cozy-bar in mobile view only', () => {
    const mockProps = getMockProps()
    const component = shallow(<Discover t={tMock} {...mockProps} />)
    expect(component.find(BarCenter).length).toBe(0)
    const componentMobile = shallow(
      <Discover t={tMock} breakpoints={{ isMobile: true }} {...mockProps} />
    )
    expect(componentMobile.find(BarCenter).length).toBe(1)
    expect(componentMobile.getElement()).toMatchSnapshot()
  })
})
