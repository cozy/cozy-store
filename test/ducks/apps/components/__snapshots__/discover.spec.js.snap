// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discover component should be rendered correctly with apps 1`] = `
<Content
  className="sto-discover"
>
  <div
    className="sto-list-container"
  >
    <div
      className="sto-discover-sections"
    >
      <QuerystringSections
        apps={
          Array [
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "label": 1,
              "name": "Collect",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://collect.cozy.mock/",
              "slug": "collect",
              "tags": Array [
                "konnector",
                "collect",
                "bills",
                "providers",
                "files",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0",
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "isp",
                "telecom",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "name": "Bouilligue",
              "slug": "konnector-bouilligue",
              "type": "konnector",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "transport",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Trinlane",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
              },
              "slug": "konnector-trinlane",
              "tags": Array [
                "transport",
                "files",
                "bills",
              ],
              "type": "konnector",
              "uninstallable": true,
              "version": "0.1.0",
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "locales": Object {
                "en": Object {
                  "long_description": "A long description finally short",
                },
              },
              "name": "Photos",
              "name_prefix": "Cozy",
              "slug": "photos",
              "type": "webapp",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
            Object {
              "availableVersion": "3.0.0",
              "categories": Array [
                "partners",
              ],
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Tasky",
              "slug": "tasky",
              "type": "webapp",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                  "1.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                  "1.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                  "1.0.0",
                ],
              },
            },
          ]
        }
        error={null}
        onAppClick={[Function]}
      />
    </div>
    <withI18n(AppVote) />
  </div>
  <withI18n(ApplicationRoutingWrapper)
    actionError={null}
    apps={
      Array [
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "label": 1,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "isp",
            "telecom",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "isInRegistry": true,
          "name": "Bouilligue",
          "slug": "konnector-bouilligue",
          "type": "konnector",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "transport",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Trinlane",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
          },
          "slug": "konnector-trinlane",
          "tags": Array [
            "transport",
            "files",
            "bills",
          ],
          "type": "konnector",
          "uninstallable": true,
          "version": "0.1.0",
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "isInRegistry": true,
          "locales": Object {
            "en": Object {
              "long_description": "A long description finally short",
            },
          },
          "name": "Photos",
          "name_prefix": "Cozy",
          "slug": "photos",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "availableVersion": "3.0.0",
          "categories": Array [
            "partners",
          ],
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Tasky",
          "slug": "tasky",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "3.0.0",
              "1.0.0",
            ],
            "dev": Array [
              "3.0.0",
              "1.0.0",
            ],
            "stable": Array [
              "3.0.0",
              "1.0.0",
            ],
          },
        },
      ]
    }
    isFetching={false}
    parent="discover"
  />
</Content>
`;

exports[`Discover component should be rendered correctly with apps and URL search params 1`] = `
<Content
  className="sto-discover"
>
  <div
    className="sto-list-container"
  >
    <div
      className="sto-discover-sections"
    >
      <QuerystringSections
        apps={
          Array [
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "label": 1,
              "name": "Collect",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://collect.cozy.mock/",
              "slug": "collect",
              "tags": Array [
                "konnector",
                "collect",
                "bills",
                "providers",
                "files",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0",
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "isp",
                "telecom",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "name": "Bouilligue",
              "slug": "konnector-bouilligue",
              "type": "konnector",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "transport",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Trinlane",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
              },
              "slug": "konnector-trinlane",
              "tags": Array [
                "transport",
                "files",
                "bills",
              ],
              "type": "konnector",
              "uninstallable": true,
              "version": "0.1.0",
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "locales": Object {
                "en": Object {
                  "long_description": "A long description finally short",
                },
              },
              "name": "Photos",
              "name_prefix": "Cozy",
              "slug": "photos",
              "type": "webapp",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
            Object {
              "availableVersion": "3.0.0",
              "categories": Array [
                "partners",
              ],
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Tasky",
              "slug": "tasky",
              "type": "webapp",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                  "1.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                  "1.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                  "1.0.0",
                ],
              },
            },
          ]
        }
        error={null}
        onAppClick={[Function]}
      />
    </div>
    <withI18n(AppVote) />
  </div>
  <withI18n(ApplicationRoutingWrapper)
    actionError={null}
    apps={
      Array [
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "label": 1,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "isp",
            "telecom",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "isInRegistry": true,
          "name": "Bouilligue",
          "slug": "konnector-bouilligue",
          "type": "konnector",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "transport",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Trinlane",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
          },
          "slug": "konnector-trinlane",
          "tags": Array [
            "transport",
            "files",
            "bills",
          ],
          "type": "konnector",
          "uninstallable": true,
          "version": "0.1.0",
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "isInRegistry": true,
          "locales": Object {
            "en": Object {
              "long_description": "A long description finally short",
            },
          },
          "name": "Photos",
          "name_prefix": "Cozy",
          "slug": "photos",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "availableVersion": "3.0.0",
          "categories": Array [
            "partners",
          ],
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Tasky",
          "slug": "tasky",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "3.0.0",
              "1.0.0",
            ],
            "dev": Array [
              "3.0.0",
              "1.0.0",
            ],
            "stable": Array [
              "3.0.0",
              "1.0.0",
            ],
          },
        },
      ]
    }
    isFetching={false}
    parent="discover"
  />
</Content>
`;

exports[`Discover component should display error from props correctly 1`] = `
<Content
  className="sto-discover"
>
  <div
    className="sto-list-container"
  >
    <div
      className="sto-discover-sections"
    >
      <QuerystringSections
        apps={Array []}
        error={[Error: This is a test error]}
        onAppClick={[Function]}
      />
    </div>
    <withI18n(AppVote) />
  </div>
  <withI18n(ApplicationRoutingWrapper)
    actionError={null}
    apps={Array []}
    isFetching={false}
    parent="discover"
  />
</Content>
`;

exports[`Discover component should render correctly a spinner if apps is fetching 1`] = `
<Content
  className="sto-discover"
>
  <div
    className="sto-list-container"
  >
    <div
      className="sto-discover-sections"
    />
  </div>
  <withI18n(ApplicationRoutingWrapper)
    actionError={null}
    apps={Array []}
    isFetching={true}
    parent="discover"
  />
</Content>
`;

exports[`Discover component should use BarCenter from cozy-bar in mobile view only 1`] = `
<Content
  className="sto-discover"
>
  <div
    className="sto-list-container"
  >
    <BarCenter>
      <h2
        className="sto-view-title"
      >
        Applications
      </h2>
    </BarCenter>
    <div
      className="sto-discover-sections"
    >
      <QuerystringSections
        apps={
          Array [
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "label": 1,
              "name": "Collect",
              "name_prefix": "Cozy",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
                "mock2": Object {
                  "type": "io.mock.doctype2",
                },
              },
              "related": "http://collect.cozy.mock/",
              "slug": "collect",
              "tags": Array [
                "konnector",
                "collect",
                "bills",
                "providers",
                "files",
              ],
              "type": "webapp",
              "uninstallable": false,
              "version": "3.0.0",
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "isp",
                "telecom",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "name": "Bouilligue",
              "slug": "konnector-bouilligue",
              "type": "konnector",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "transport",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Trinlane",
              "permissions": Object {
                "mock": Object {
                  "type": "io.mock.doctype",
                },
              },
              "slug": "konnector-trinlane",
              "tags": Array [
                "transport",
                "files",
                "bills",
              ],
              "type": "konnector",
              "uninstallable": true,
              "version": "0.1.0",
              "versions": Object {
                "beta": Array [
                  "0.1.0",
                ],
                "dev": Array [
                  "0.1.0",
                ],
                "stable": Array [
                  "0.1.0",
                ],
              },
            },
            Object {
              "categories": Array [
                "cozy",
              ],
              "developer": Object {
                "name": "Cozy",
              },
              "editor": "Cozy",
              "icon": "<svg></svg>",
              "isInRegistry": true,
              "locales": Object {
                "en": Object {
                  "long_description": "A long description finally short",
                },
              },
              "name": "Photos",
              "name_prefix": "Cozy",
              "slug": "photos",
              "type": "webapp",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                ],
              },
            },
            Object {
              "availableVersion": "3.0.0",
              "categories": Array [
                "partners",
              ],
              "icon": "<svg></svg>",
              "installed": true,
              "isInRegistry": true,
              "name": "Tasky",
              "slug": "tasky",
              "type": "webapp",
              "uninstallable": true,
              "versions": Object {
                "beta": Array [
                  "3.0.0",
                  "1.0.0",
                ],
                "dev": Array [
                  "3.0.0",
                  "1.0.0",
                ],
                "stable": Array [
                  "3.0.0",
                  "1.0.0",
                ],
              },
            },
          ]
        }
        error={null}
        onAppClick={[Function]}
      />
    </div>
    <withI18n(AppVote) />
  </div>
  <withI18n(ApplicationRoutingWrapper)
    actionError={null}
    apps={
      Array [
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "label": 1,
          "name": "Collect",
          "name_prefix": "Cozy",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
            "mock2": Object {
              "type": "io.mock.doctype2",
            },
          },
          "related": "http://collect.cozy.mock/",
          "slug": "collect",
          "tags": Array [
            "konnector",
            "collect",
            "bills",
            "providers",
            "files",
          ],
          "type": "webapp",
          "uninstallable": false,
          "version": "3.0.0",
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "isp",
            "telecom",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "isInRegistry": true,
          "name": "Bouilligue",
          "slug": "konnector-bouilligue",
          "type": "konnector",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "transport",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Trinlane",
          "permissions": Object {
            "mock": Object {
              "type": "io.mock.doctype",
            },
          },
          "slug": "konnector-trinlane",
          "tags": Array [
            "transport",
            "files",
            "bills",
          ],
          "type": "konnector",
          "uninstallable": true,
          "version": "0.1.0",
          "versions": Object {
            "beta": Array [
              "0.1.0",
            ],
            "dev": Array [
              "0.1.0",
            ],
            "stable": Array [
              "0.1.0",
            ],
          },
        },
        Object {
          "categories": Array [
            "cozy",
          ],
          "developer": Object {
            "name": "Cozy",
          },
          "editor": "Cozy",
          "icon": "<svg></svg>",
          "isInRegistry": true,
          "locales": Object {
            "en": Object {
              "long_description": "A long description finally short",
            },
          },
          "name": "Photos",
          "name_prefix": "Cozy",
          "slug": "photos",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "3.0.0",
            ],
            "dev": Array [
              "3.0.0",
            ],
            "stable": Array [
              "3.0.0",
            ],
          },
        },
        Object {
          "availableVersion": "3.0.0",
          "categories": Array [
            "partners",
          ],
          "icon": "<svg></svg>",
          "installed": true,
          "isInRegistry": true,
          "name": "Tasky",
          "slug": "tasky",
          "type": "webapp",
          "uninstallable": true,
          "versions": Object {
            "beta": Array [
              "3.0.0",
              "1.0.0",
            ],
            "dev": Array [
              "3.0.0",
              "1.0.0",
            ],
            "stable": Array [
              "3.0.0",
              "1.0.0",
            ],
          },
        },
      ]
    }
    isFetching={false}
    parent="discover"
  />
</Content>
`;
